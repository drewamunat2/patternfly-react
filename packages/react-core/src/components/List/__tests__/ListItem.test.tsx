import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { ListItem } from '../ListItem';
import BookOpenIcon from '@patternfly/react-icons/dist/esm/icons/book-open-icon';

test('Renders children', () => {
  render(<ListItem>Test</ListItem>);
  expect(screen.getByText('Test')).toBeVisible();
});

test('Renders with the class pf-c-list__item when icon prop is passed', () => {
  render(<ListItem icon={<BookOpenIcon />}>Test</ListItem>);
  expect(screen.getByText('Test')).toHaveClass('pf-c-list__item');
});

test('Matches the snapshot with span when icon prop is passed', () => {
  const { asFragment } = render(<ListItem icon={<BookOpenIcon />}>Test</ListItem>);
  expect(asFragment()).toMatchSnapshot();
});

test('Matches the snapshot without span by default', () => {
  const { asFragment } = render(<ListItem>Test</ListItem>);
  expect(asFragment()).toMatchSnapshot();
});
