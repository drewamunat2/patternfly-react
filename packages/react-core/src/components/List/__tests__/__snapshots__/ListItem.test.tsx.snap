// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot with span when icon prop is passed 1`] = `
<DocumentFragment>
  <li
    class="pf-c-list__item"
  >
    <span
      class="pf-c-list__item-icon"
    >
      icon
    </span>
    Test
  </li>
</DocumentFragment>
`;

exports[`Matches the snapshot without span by default 1`] = `
<DocumentFragment>
  <li
    class=""
  >
    Test
  </li>
</DocumentFragment>
`;
